lib/usd/utils/MergePrims.cpp

```diff
@@ -159,6 +161,138 @@ template <class T> auto makeFuncWithContext(const MergeContext& ctx, T&& func)
         func, ctx, ph::_1, ph::_2, ph::_3, ph::_4, ph::_5, ph::_6, ph::_7, ph::_8, ph::_9);
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+// Special transform attributes handling.
+//
+// The goal is to detect that a prim transform has *not* changed even though its representation
+// might have changed. For example it might use different attributes (RotateX vs RotateXYZ) or
+// different transform operations order. (These are in the xformOpOrder attribute.)
+//----------------------------------------------------------------------------------------------------------------------
+
+const std::set<TfToken>& getTransformAttributeNames()
```

Comment:
Ouch, I did not know about that. It seems it will complicate things even further. OTOH, we can document that we only support a given subset as a best effort. If you have ideas how to make it work better, I'm all ears. Maybe there is a USD function that can identify if an attribute is a transform attribute?